android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java

```diff
@@ -117,26 +105,10 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result
     }
 
     private void register(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
-        final Long handle = _nextRegistration++;
-
-        if (null != _ablyHandle) {
-            System.out.println("Disposing of previous Ably instance (# " + _ablyHandle + ").");
-        }
-
-        // Setting _ablyHandle to null when _ably is not null indicates that we're in the process
-        // of asynchronously disposing of the old instance.
-        _ablyHandle = null;
-
-        // TODO actually do this next bit asynchronously like we do for iOS
-        if (null != _ably) {
-            _ably.dispose();
-        }
-
-        System.out.println("Creating new Ably instance (# " + handle + ").");
-        _ably = new AblyLibrary();
-        _ablyHandle = handle;
-
-        result.success(handle);
+        System.out.println("Registering library instance to clean up any existing instnaces");
+        onHotRestartListener.trigger();
```

Comment:
Where is this implemented?... I can't seem to work out how this works.

Maybe my `grep` skills are lacking?

```
ably-flutter % git grep trigger
README.md:As per RTE6a, 2nd listener should also be triggered, but it will not be as 2nd listener is registered after 1st listener and  stream subscription is cancelled immediately after first listener is triggered.
android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java:        void trigger();
android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java:        onHotRestartListener.trigger();
example/lib/main.dart:          await preZetaSubscription.cancel();  //by the time this cancel is triggered, preZeta will already have received current event.
example/lib/main.dart:          await postZetaSubscription.cancel(); //by the time this cancel is triggered, postZeta hasn't received the event yet. And will never receive as it is cancelled.
example/test/widget_test.dart:    // Build our app and trigger a frame.
ably-flutter % git grep OnHotRestart
android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java:    public interface OnHotRestart {
android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java:    private OnHotRestart onHotRestartListener;
android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java:    static synchronized AblyMethodCallHandler getInstance(OnHotRestart listener) {
android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java:    private AblyMethodCallHandler(OnHotRestart listener) {
```

And, also, the log statement has a typo - "instances".

Reply:
@QuintinWillison your grep skills are at 💯  
Just that we are using the not-so-obvious lambda expression (here, method reference, actually) [in AblyFlutterPlugin.java](https://github.com/ably/ably-flutter/blob/feature/synchronous-specification/android/src/main/java/io/ably/flutter/plugin/AblyFlutterPlugin.java#L41) 😉

Reply:
ok, thanks.

If there's anything you can do to either add commentary explaining that, or better provide some level of interim expression in code (e.g. interim typedef if that's a thing) then that might help. I'm thinking future maintainability - you, me, anybody else coming to this code in months or years time.

Reply:
Not sure if that is required! I feel the code is explanatory as you can see the usage of definition in below line form your logs...

```
android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java:    static synchronized AblyMethodCallHandler getInstance(OnHotRestart listener) {
```

Reply:
ok, just bear in mind that maintainers of this codebase in the future will likely be short of time so we need to make things obvious wherever possible. If you feel this is obvious enough then leave it be. I just wanted to point out that I found this particular part of the event flow difficult to follow.